tests/agents/gherkin-specs/outcome.feature

```diff
@@ -0,0 +1,110 @@
+    And span terminates without error
+    Then span outcome is 'success'
+
+  Scenario: transaction with error
+    Given an agent
+    And an active transaction
+    And transaction terminates with an error
+    Then transaction outcome is 'failure'
+
+  Scenario: transaction without error
+    Given an agent
+    And an active transaction
+    And transaction terminates without error
+    Then transaction outcome is 'success'
+
+  # ---- HTTP
+
+  @http
+  Scenario Outline: HTTP transaction and span outcome
+    Given an agent
+    And an HTTP transaction with <status> response code
+    Then transaction outcome is "<server>"
+    Given an HTTP span with <status> response code
+    Then span outcome is "<client>"
+    Examples:
+      | status | client  | server  |
+      | 100    | success | success |
+      | 200    | success | success |
+      | 300    | success | success |
+      | 400    | failure | success |
+      | 404    | failure | success |
+      | 500    | failure | failure |
+      | -1     | failure | failure |
+      # last row with negative status represents the case where the status is not available
+      # for example when an exception/error is thrown without status (IO error, redirect loop, ...)
+
+  # ---- gRPC
+
+  # reference spec : https://github.com/grpc/grpc/blob/master/doc/statuscodes.md
+  #
+  # The following statuses are not used by gRPC client & server,
+  # thus they should be considered as client-side errors
+  #
+  # - INVALID_ARGUMENT
+  # - NOT_FOUND
+  # - ALREADY_EXISTS
+  # - PERMISSION_DENIED
+  # - FAILED_PRECONDITION
+  # - ABORTED
+  # - OUT_OF_RANGE
+  # - DATA_LOSS
+  # - UNAUTHENTICATED
+
+  @grpc
+  Scenario Outline: gRPC transaction and span outcome
+    Given an agent
+    And a gRPC transaction with '<status>' status
+    Then transaction outcome is "<server>"
+    Given a gRPC span with '<status>' status
+    Then span outcome is "<client>"
+    Examples:
+      | status              | client  | server  |
+      | OK                  | success | success |
+      | CANCELLED           | failure | failure |
+      | UNKNOWN             | failure | failure |
+      | INVALID_ARGUMENT    | failure | success |
+      | DEADLINE_EXCEEDED   | failure | failure |
+      | NOT_FOUND           | failure | success |
+      | ALREADY_EXISTS      | failure | success |
+      | PERMISSION_DENIED   | failure | success |
+      | RESOURCE_EXHAUSTED  | failure | failure |
+      | FAILED_PRECONDITION | failure | success |
+      | ABORTED             | failure | success |
+      | OUT_OF_RANGE        | failure | success |
+      | UNIMPLEMENTED       | failure | failure |
+      | INTERNAL            | failure | failure |
+      | UNAVAILABLE         | failure | failure |
+      | DATA_LOSS           | failure | success |
+      | UNAUTHENTICATED     | failure | success |
+      | n/a                 | failure | failure |
+    # last row with 'n/a' status represents the case where status is not available
```

Comment:
I think these still require update to be in sync with the gRPC spec

Reply:
Good catch, I forgot to update those !